### PR TITLE
jupyter notebooks from examples

### PR DESCRIPTION
**Pull request by arokem**
This continues the work in #1022 and supersedes it.

### CODE REVIEW

**Comment by arokem** [comment]
Still needs some more cleaning up. The rendered notebooks will include (for example) figure blocks from the original rst and other crufy, so we need to process the cells even further. Will follow up with that, before marking this as "ready for review"

File: tools/make_notebooks.py [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+    textcells = [new_markdown_cell(source=textcells[i])
+                 for i in range(len(textcells))]
+
+    cells = []
+    for i in range(0, len(allcells)):
+        try:
+            cells.append(textcells[i])
+            cells.append(codecells[i])
+        except:
+            pass
```
Comment (Borda) [inline tools/make_notebooks.py]: just `pass` is not best practice... :)

File: tools/make_notebooks.py [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+                call the make_notebook function
+
+                Default to ../doc/examples/
+    Returns
+    -------
+    None
+    """
+
+    file_path = os.path.join(directory, fname)
+    f = open(file_path, "r")
```
Comment (Borda) [inline tools/make_notebooks.py]: rather use `with` statement
```
with open(file_path, 'r') as fp:
    fdata = fp.read()
```

File: tools/make_notebooks.py [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+
+    nbf.write(notebook, nbname, 4)
+    nbname.close()
+
+
+def valid_examples():
+    """Get the valid examples to be converted"""
+    flist_name = pjoin(os.path.dirname(os.getcwd()), 'doc', 'examples',
+                       'valid_examples.txt')
+    flist = open(flist_name, "r")
```
Comment (Borda) [inline tools/make_notebooks.py]: again, rather use `with` statement

File: tools/make_notebooks.py [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+    validated_examples = [line.split("#", 1)[0] for line in validated_examples]
+    # Remove leading and trailing white space from example names
+    validated_examples = [line.strip() for line in validated_examples]
+    # Remove blank lines
+    validated_examples = filter(None, validated_examples)
+    out_examples = []
+    for example in validated_examples:
+        fullpath = pjoin(EG_SRC_DIR, example)
+        if not example.endswith(".py"):
+            print("%s not a python file, skipping." % example)
```
Comment (Borda) [inline tools/make_notebooks.py]: why not using `logging` module?

**Pull request closed without merging**